internal/tracker/debounced_queue.go

```diff
@@ -0,0 +1,196 @@
+package tracker
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// DebouncedQueue provides a queue with built-in debouncing functionality
+// It encapsulates both debouncing logic and workqueue implementation
+type DebouncedQueue interface {
```

Comment:
Could you please explain why we need this queue and is there some risk of data corruption when the operator gets restarted and we lose the state?

Reply:
The queue has two purposes
- debounce the operations, so we don't risk flooding the API server with requests
- handle asynchronous retries, should there be any issues with updating the model validation resources

@osmman [asked a similar question](https://github.com/sigstore/model-validation-operator/pull/47#discussion_r2260334208) about restarts